core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -538,48 +551,57 @@ class GroupMetadataManager(val brokerId: Int,
         offsetAndMetadata.expireTimestamp < startMs
       }
 
-      debug("Found %d expired offsets.".format(expiredOffsets.size))
+      deleteExpiredOffsets(expiredOffsets)
+    }
+
+    info("Removed %d expired offsets in %d milliseconds.".format(numExpiredOffsetsRemoved, SystemTime.milliseconds - startMs))
+  }
+
+  /**
+   * Deletes the provided expired offsets.
+   * @param expiredOffsets collection of expired offsets.
+   * @return number of expired offsets deleted.
+   */
+  def deleteExpiredOffsets(expiredOffsets: scala.Iterable[(GroupTopicPartition, OffsetAndMetadata)]): Int = {
```

Comment:
nit: maybe rename to `deleteOffsets` since the offsets removed when a topic is deleted are not really "expired"?